Cargo.toml

```diff
@@ -19,7 +19,6 @@
 members = [
     "datafusion/common",
     "datafusion/core",
-    "datafusion/data-access",
```

Comment:
this code is unused, correct? And this is a cleanup that is not strictly necessary for this PR? (this is fine, I am just trying to make sure I understand)

Reply:
Yes.  This PR is based on another PR #2904